src/components/organisms/SchedulePageModal/SchedulePageModal.utils.ts

```diff
@@ -0,0 +1,97 @@
+import {
+  addDays,
+  compareAsc,
+  differenceInMinutes,
+  endOfDay,
+  format,
+  isWithinInterval,
+  startOfDay,
+} from "date-fns";
+
+import { Room } from "types/rooms";
+import { PersonalizedVenueEvent, VenueEvent } from "types/venues";
+import { MyPersonalizedSchedule } from "types/User";
+
+import { WithVenueId } from "utils/id";
+import { ONE_MINUTE_IN_SECONDS, ONE_SECOND_IN_MILLISECONDS } from "utils/time";
+import { isTruthy } from "utils/types";
+
+import { RoomWithEvents } from "components/molecules/Schedule/Schedule.types";
+
+export const isEventThisDay = (date: Date) => {
+  return (event: VenueEvent) => {
+    return isWithinInterval(date, {
+      start: startOfDay(event.start_utc_seconds * ONE_SECOND_IN_MILLISECONDS),
+      end: endOfDay(
+        (event.start_utc_seconds +
+          event.duration_minutes * ONE_MINUTE_IN_SECONDS) *
+          ONE_SECOND_IN_MILLISECONDS
+      ),
+    });
+  };
+};
+
+export const adjustEventTiming = (date: Date) => {
+  return (event: WithVenueId<VenueEvent>) => {
+    const dates = [
+      event.start_utc_seconds * ONE_SECOND_IN_MILLISECONDS,
+      (event.start_utc_seconds +
+        event.duration_minutes * ONE_MINUTE_IN_SECONDS) *
+        ONE_SECOND_IN_MILLISECONDS,
+      startOfDay(date),
+      endOfDay(date),
+    ].sort(compareAsc);
+
+    const startUtcMilliseconds =
+      dates[1] instanceof Date ? dates[1].getTime() : dates[1];
+
+    return {
+      ...event,
+      start_utc_seconds: startUtcMilliseconds / ONE_SECOND_IN_MILLISECONDS,
+      duration_minutes: differenceInMinutes(dates[2], dates[1]),
+    };
+  };
+};
```

Comment:
This seems super verbose/complex.. what's the intended functionality here? I feel like we can probably simplify it a lot.

Reply:
oh yeah... I knew that but couldn't come up with anything better before today.
i changed structure a bit and changed the logic. Now It's functionality is in `prepareForSchedule` https://github.com/sparkletown/sparkle/pull/1302/commits/9fce282d69bf3236a3cad6381d0b990072e0c3c2